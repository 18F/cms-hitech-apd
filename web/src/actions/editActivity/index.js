export * from './contractorResources';
export * from './costAllocate';
export * from './overview';
