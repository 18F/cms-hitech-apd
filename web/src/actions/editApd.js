import { updateBudget } from './apd';

export const ADD_APD_ITEM = Symbol('add apd item');
export const EDIT_APD = Symbol('edit apd');
export const REMOVE_APD_ITEM = Symbol('remove apd item');

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};

export const setMedicaidDirectorEmail = email => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/email',
    value: email
  };
};

export const setMedicaidDirectorName = name => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/name',
    value: name
  };
};

export const setMedicaidDirectorPhoneNumber = phone => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/phone',
    value: phone
  };
};

export const setMedicaidOfficeAddress1 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address1',
    value: address
  };
};

export const setMedicaidOfficeAddress2 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address2',
    value: address
  };
};

export const setMedicaidOfficeCity = city => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/city',
    value: city
  };
};

export const setMedicaidOfficeState = state => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/state',
    value: state
  };
};

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
