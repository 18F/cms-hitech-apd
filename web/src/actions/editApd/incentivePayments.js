import { EDIT_APD } from './symbols';

export const setIncentiveEHCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehCt/${year}/${quarter}`,
  value
});

export const setIncentiveEHPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehAmt/${year}/${quarter}`,
  value
});

export const setIncentiveEPCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epCt/${year}/${quarter}`,
  value
});

export const setIncentiveEPPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epAmt/${year}/${quarter}`,
  value
});
