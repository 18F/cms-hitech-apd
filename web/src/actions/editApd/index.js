export * from './symbols';

export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
