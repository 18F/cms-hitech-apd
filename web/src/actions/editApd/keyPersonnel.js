import { ADD_APD_ITEM, EDIT_APD, REMOVE_APD_ITEM } from './symbols';
import { updateBudget } from '../apd';

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};
