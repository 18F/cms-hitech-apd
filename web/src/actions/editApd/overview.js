import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
import { updateBudget } from '../apd';

export const addYear = year => (dispatch, getState) => {
  dispatch({ type: ADD_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const removeYear = year => (dispatch, getState) => {
  dispatch({ type: REMOVE_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const setNarrativeForHIE = text => ({
  type: EDIT_APD,
  path: '/narrativeHIE',
  value: text
});

export const setNarrativeForHIT = text => ({
  type: EDIT_APD,
  path: '/narrativeHIT',
  value: text
});

export const setNarrativeForMMIS = text => ({
  type: EDIT_APD,
  path: '/narrativeMMIS',
  value: text
});

export const setProgramOverview = text => ({
  type: EDIT_APD,
  path: '/programOverview',
  value: text
});
