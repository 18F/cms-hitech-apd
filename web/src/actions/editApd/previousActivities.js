import { EDIT_APD } from './symbols';

export const setPreviousActivitySummary = summary => ({
  type: EDIT_APD,
  path: '/previousActivitySummary',
  value: summary
});

export const setPreviousActivityApprovedExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseForHITandHIE = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/hithie/federalActual`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/totalApproved`,
  value: expense
});

export const setPreviousActivityApprovedExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/totalApproved`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS50FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/50/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS75FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/75/federalActual`,
  value: expense
});

export const setPreviousActivityFederalActualExpenseforMMIS90FFP = (
  year,
  expense
) => ({
  type: EDIT_APD,
  path: `/previousActivityExpenses/${year}/mmis/90/federalActual`,
  value: expense
});
