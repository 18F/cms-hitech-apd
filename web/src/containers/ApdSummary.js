import React from 'react';
import { useDispatch, useSelector } from 'react-redux';

import Waypoint from './ConnectedWaypoint';
import {
  addYear,
  removeYear,
  setNarrativeForHIE,
  setNarrativeForHIT,
  setNarrativeForMMIS,
  setProgramOverview
} from '../actions/editApd';
import Choice from '../components/Choice';
import RichText from '../components/RichText';
import Instruction from '../components/Instruction';
import { Section } from '../components/Section';
import { t } from '../i18n';

const ApdSummary = () => {
  const dispatch = useDispatch();

  const {
    narrativeHIE,
    narrativeHIT,
    narrativeMMIS,
    programOverview,
    years,
    yearOptions
  } = useSelector(({ apd: { data } }) => data);

  const handleYears = e => {
    const { value } = e.target;
    if (years.includes(value)) {
      dispatch(removeYear(value));
    } else {
      dispatch(addYear(value));
    }
  };

  const syncRichText = action => html => {
    dispatch(action(html));
  };

  return (
    <Waypoint id="apd-summary">
      <Section isNumbered id="apd-summary" resource="apd">
        <h3 className="ds-h3 subsection--title">{t('apd.overview.title')}</h3>
        <Instruction source="apd.overview.instruction" />

        {yearOptions.map(option => (
          <Choice
            key={option}
            checked={years.includes(option)}
            name={`apd year ${option}`}
            onChange={handleYears}
            type="checkbox"
            value={option}
          >
            {option}
          </Choice>
        ))}

        <div className="ds-u-margin-y--3">
          <Instruction source="apd.introduction.instruction" />
          <RichText
            content={programOverview}
            onSync={syncRichText(setProgramOverview)}
            editorClassName="rte-textarea-l"
          />
        </div>
        <div className="ds-u-margin-bottom--3">
          <Instruction source="apd.hit.instruction" />
          <RichText
            content={narrativeHIT}
            onSync={syncRichText(setNarrativeForHIT)}
            editorClassName="rte-textarea-l"
          />
        </div>
        <div className="ds-u-margin-bottom--3">
          <Instruction source="apd.hie.instruction" />
          <RichText
            content={narrativeHIE}
            onSync={syncRichText(setNarrativeForHIE)}
            editorClassName="rte-textarea-l"
          />
        </div>
        <div>
          <Instruction source="apd.mmis.instruction" />
          <RichText
            content={narrativeMMIS}
            onSync={syncRichText(setNarrativeForMMIS)}
            editorClassName="rte-textarea-l"
          />
        </div>
      </Section>
    </Waypoint>
  );
};

export default ApdSummary;
