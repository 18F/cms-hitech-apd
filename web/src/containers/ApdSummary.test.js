import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import ApdSummary from './ApdSummary';
import { UPDATE_BUDGET } from '../actions/apd';
import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from '../actions/editApd';

const mockStore = configureStore([thunk]);

jest.mock('../components/RichText');

describe('apd summary component', () => {
  const state = {
    apd: {
      data: {
        narrativeHIE: 'about hie',
        narrativeHIT: 'about hit',
        narrativeMMIS: 'about mmis',
        programOverview: 'about the program',
        yearOptions: ['1', '2', '3'],
        years: ['1', '2']
      }
    }
  };
  const store = mockStore(state);

  beforeEach(() => {
    store.clearActions();
  });

  test('renders correctly', () => {
    expect(
      mount(
        <Provider store={store}>
          <ApdSummary />
        </Provider>
      )
    ).toMatchSnapshot();
  });

  test('dispatches on text change', () => {
    mount(
      <Provider store={store}>
        <ApdSummary />
      </Provider>
    )
      .find('RichText')
      .at(0)
      .prop('onSync')('this is some html');

    // this one is based on knowledge that the program overview comes first
    expect(store.getActions()).toEqual([
      {
        type: EDIT_APD,
        path: '/programOverview',
        value: 'this is some html'
      }
    ]);
  });

  test('dispatches when adding a year', () => {
    // add a year
    mount(
      <Provider store={store}>
        <ApdSummary />
      </Provider>
    )
      .find('ChoiceComponent[value="3"]')
      .prop('onChange')({ target: { value: '3' } });
    expect(store.getActions()).toEqual([
      { type: ADD_APD_YEAR, value: '3', state },
      { type: UPDATE_BUDGET, state }
    ]);
  });

  test('dispatches when removing a year', () => {
    // remove a year
    mount(
      <Provider store={store}>
        <ApdSummary />
      </Provider>
    )
      .find('ChoiceComponent[value="2"]')
      .prop('onChange')({ target: { value: '2' } });
    expect(store.getActions()).toEqual([
      { type: REMOVE_APD_YEAR, value: '2', state },
      { type: UPDATE_BUDGET, state }
    ]);
  });
});
