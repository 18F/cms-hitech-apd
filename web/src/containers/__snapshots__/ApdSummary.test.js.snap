// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apd summary component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ApdSummary>
    <Connect(ConnectedWaypoint)
      id="apd-summary"
    >
      <ConnectedWaypoint
        id="apd-summary"
        scrollTo={[Function]}
      >
        <Waypoint
          bottomOffset="90%"
          debug={false}
          fireOnRapidScroll={true}
          horizontal={false}
          onEnter={[Function]}
          onLeave={[Function]}
          onPositionChange={[Function]}
          topOffset="0px"
        >
          <span
            style={
              Object {
                "fontSize": 0,
              }
            }
          />
        </Waypoint>
        <Section
          id="apd-summary"
          isNumbered={true}
          resource="apd"
        >
          <section
            className="numbered-section"
            id="apd-summary"
          >
            <h2
              className="ds-h2"
            >
              Program Summary
            </h2>
            <span
              className="ds-text--lead"
            >
              Provide an introduction to the state program. Include context for this funding request by providing information such as current size, future vision, and how the program is positioned within the state. This information is similar to the content from the IAPD executive summary template section.
            </span>
            <h3
              className="ds-h3 subsection--title"
            >
              Program Overview
            </h3>
            <Instruction
              args={null}
              headingDisplay={
                Object {
                  "className": "ds-h3",
                  "level": "h3",
                }
              }
              reverse={false}
              source="apd.overview.instruction"
            >
              <div>
                <div
                  className="visibility--screen"
                >
                  <p
                    className="ds-u-font-weight--bold"
                  >
                    Choose the federal fiscal year(s) this APD covers.
                  </p>
                </div>
              </div>
            </Instruction>
            <ChoiceComponent
              checked={true}
              key="1"
              name="apd year 1"
              onChange={[Function]}
              type="checkbox"
              value="1"
            >
              <div>
                <Choice
                  checked={true}
                  inputPlacement="left"
                  name="apd year 1"
                  onChange={[Function]}
                  type="checkbox"
                  value="1"
                >
                  <div
                    aria-atomic={null}
                    aria-live={null}
                    aria-relevant={null}
                  >
                    <input
                      checked={true}
                      className="ds-c-choice"
                      id="checkbox_apd year 1_1"
                      name="apd year 1"
                      onChange={[Function]}
                      type="checkbox"
                      value="1"
                    />
                    <FormLabel
                      component="label"
                      fieldId="checkbox_apd year 1_1"
                    >
                      <label
                        className="ds-c-label"
                        htmlFor="checkbox_apd year 1_1"
                      >
                        <span
                          className=""
                        >
                          1
                        </span>
                      </label>
                    </FormLabel>
                  </div>
                </Choice>
              </div>
            </ChoiceComponent>
            <ChoiceComponent
              checked={true}
              key="2"
              name="apd year 2"
              onChange={[Function]}
              type="checkbox"
              value="2"
            >
              <div>
                <Choice
                  checked={true}
                  inputPlacement="left"
                  name="apd year 2"
                  onChange={[Function]}
                  type="checkbox"
                  value="2"
                >
                  <div
                    aria-atomic={null}
                    aria-live={null}
                    aria-relevant={null}
                  >
                    <input
                      checked={true}
                      className="ds-c-choice"
                      id="checkbox_apd year 2_2"
                      name="apd year 2"
                      onChange={[Function]}
                      type="checkbox"
                      value="2"
                    />
                    <FormLabel
                      component="label"
                      fieldId="checkbox_apd year 2_2"
                    >
                      <label
                        className="ds-c-label"
                        htmlFor="checkbox_apd year 2_2"
                      >
                        <span
                          className=""
                        >
                          2
                        </span>
                      </label>
                    </FormLabel>
                  </div>
                </Choice>
              </div>
            </ChoiceComponent>
            <ChoiceComponent
              checked={false}
              key="3"
              name="apd year 3"
              onChange={[Function]}
              type="checkbox"
              value="3"
            >
              <div>
                <Choice
                  checked={false}
                  inputPlacement="left"
                  name="apd year 3"
                  onChange={[Function]}
                  type="checkbox"
                  value="3"
                >
                  <div
                    aria-atomic={null}
                    aria-live={null}
                    aria-relevant={null}
                  >
                    <input
                      checked={false}
                      className="ds-c-choice"
                      id="checkbox_apd year 3_3"
                      name="apd year 3"
                      onChange={[Function]}
                      type="checkbox"
                      value="3"
                    />
                    <FormLabel
                      component="label"
                      fieldId="checkbox_apd year 3_3"
                    >
                      <label
                        className="ds-c-label"
                        htmlFor="checkbox_apd year 3_3"
                      >
                        <span
                          className=""
                        >
                          3
                        </span>
                      </label>
                    </FormLabel>
                  </div>
                </Choice>
              </div>
            </ChoiceComponent>
            <div
              className="ds-u-margin-y--3"
            >
              <Instruction
                args={null}
                headingDisplay={
                  Object {
                    "className": "ds-h3",
                    "level": "h3",
                  }
                }
                reverse={false}
                source="apd.introduction.instruction"
              >
                <div>
                  <h3
                    className="ds-h3"
                  >
                    Program Introduction
                  </h3>
                  <div
                    className="visibility--screen"
                  >
                    <Md
                      content="Provide a high-level introduction to the state program. Include the history and context of the program and the state’s future vision for it."
                      wrapper="p"
                    >
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Provide a high-level introduction to the state program. Include the history and context of the program and the state’s future vision for it.",
                          }
                        }
                      />
                    </Md>
                  </div>
                </div>
              </Instruction>
              <RichText
                content="about the program"
                editorClassName="rte-textarea-l"
                onSync={[Function]}
              />
            </div>
            <div
              className="ds-u-margin-bottom--3"
            >
              <Instruction
                args={null}
                headingDisplay={
                  Object {
                    "className": "ds-h3",
                    "level": "h3",
                  }
                }
                reverse={false}
                source="apd.hit.instruction"
              >
                <div>
                  <h3
                    className="ds-h3"
                  >
                    HIT Overview
                  </h3>
                  <div
                    className="visibility--screen"
                  >
                    <Md
                      content="Briefly summarize planned HIT-funded activities contained within this APD."
                      wrapper="p"
                    >
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Briefly summarize planned HIT-funded activities contained within this APD.",
                          }
                        }
                      />
                    </Md>
                  </div>
                </div>
              </Instruction>
              <RichText
                content="about hit"
                editorClassName="rte-textarea-l"
                onSync={[Function]}
              />
            </div>
            <div
              className="ds-u-margin-bottom--3"
            >
              <Instruction
                args={null}
                headingDisplay={
                  Object {
                    "className": "ds-h3",
                    "level": "h3",
                  }
                }
                reverse={false}
                source="apd.hie.instruction"
              >
                <div>
                  <h3
                    className="ds-h3"
                  >
                    HIE Overview
                  </h3>
                  <div
                    className="visibility--screen"
                  >
                    <Md
                      content="Summarize planned HIE-funded activities within this APD.
If there are no HIE activities, enter “No HIE activities.”

[ol] Description of the state’s HIE approach (statewide, sub-state health information organization (HIO), etc.);
Anticipated risks and mitigation strategies;
The short and long term value proposition to providers;
The role of state government in governance and policy-setting. [/ol]"
                      wrapper="p"
                    >
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Summarize planned HIE-funded activities within this APD.<br />
If there are no HIE activities, enter “No HIE activities.”<br />
<br />
<ol><li>Description of the state’s HIE approach (statewide, sub-state health information organization (HIO), etc.);</li><li>Anticipated risks and mitigation strategies;</li><li>The short and long term value proposition to providers;</li><li>The role of state government in governance and policy-setting.</li></ol>",
                          }
                        }
                      />
                    </Md>
                    <Md
                      className="instruction-box"
                      content="CMS recommends that states connect with regional CMS HITECH contacts prior to developing an HIE funding request. HIE funding requests are complex. Early consultation with CMS facilitates a more efficient approval process."
                      wrapper="p"
                    >
                      <p
                        className="instruction-box"
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "CMS recommends that states connect with regional CMS HITECH contacts prior to developing an HIE funding request. HIE funding requests are complex. Early consultation with CMS facilitates a more efficient approval process.",
                          }
                        }
                      />
                    </Md>
                  </div>
                </div>
              </Instruction>
              <RichText
                content="about hie"
                editorClassName="rte-textarea-l"
                onSync={[Function]}
              />
            </div>
            <div>
              <Instruction
                args={null}
                headingDisplay={
                  Object {
                    "className": "ds-h3",
                    "level": "h3",
                  }
                }
                reverse={false}
                source="apd.mmis.instruction"
              >
                <div>
                  <h3
                    className="ds-h3"
                  >
                    MMIS Overview
                  </h3>
                  <div
                    className="visibility--screen"
                  >
                    <Md
                      content="Briefly summarize planned MMIS-funded activities contained within this APD.
If there are no MMIS activities, enter “No MMIS activities.”"
                      wrapper="p"
                    >
                      <p
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "Briefly summarize planned MMIS-funded activities contained within this APD.<br />
If there are no MMIS activities, enter “No MMIS activities.”",
                          }
                        }
                      />
                    </Md>
                  </div>
                </div>
              </Instruction>
              <RichText
                content="about mmis"
                editorClassName="rte-textarea-l"
                onSync={[Function]}
              />
            </div>
          </section>
        </Section>
      </ConnectedWaypoint>
    </Connect(ConnectedWaypoint)>
  </ApdSummary>
</Provider>
`;
